apps/examples/testcase/ta_tc/wifi_manager/utc/utc_wifi_manager_main.c

```diff
@@ -0,0 +1,244 @@
+
+	TC_ASSERT_EQ("wifi_manager_set_mode", ret, WIFI_MANAGER_SUCCESS);
+
+	ret = wifi_manager_set_mode(STA_MODE, NULL);
+
+	TC_ASSERT_EQ("wifi_manager_set_mode", ret, WIFI_MANAGER_SUCCESS);
+
+	TC_SUCCESS_RESULT();
+}
+
+static void utc_wifi_manager_get_mode_n(void)
+{
+	wifi_manager_result_e ret = WIFI_MANAGER_FAIL;
+	wifi_manager_info_s *info = NULL;
+
+	ret = wifi_manager_get_info(info);
+
+	TC_ASSERT_EQ("wifi_manager_get_mode", ret, WIFI_MANAGER_FAIL);
+	TC_SUCCESS_RESULT();
+}
+
+static void utc_wifi_manager_get_mode_p(void)
+{
+	wifi_manager_result_e ret = WIFI_MANAGER_FAIL;
+	wifi_manager_info_s info;
+	int ret_cmp = 0;
+
+	ret = wifi_manager_get_info(&info);
+
+	TC_ASSERT_EQ("wifi_manager_get_mode", ret, WIFI_MANAGER_SUCCESS);
+	TC_ASSERT_EQ("wifi_manager_get_mode", info.mode, STA_MODE);
+
+	wifi_manager_softap_config_s ap_config = {"PKES", 1, "1111122222"};
+	ret = wifi_manager_set_mode(SOFTAP_MODE, &ap_config);
+
+	TC_ASSERT_EQ("wifi_manager_get_mode", ret, WIFI_MANAGER_SUCCESS);
+
+	ret = wifi_manager_get_info(&info);
+
+	TC_ASSERT_EQ("wifi_manager_get_mode", ret, WIFI_MANAGER_SUCCESS);
+	TC_ASSERT_EQ("wifi_manager_get_mode", info.mode, SOFTAP_MODE);
+	ret_cmp = strncmp(info.ssid, ap_config.ssid, sizeof(ap_config.ssid));
+	TC_ASSERT_EQ("wifi_manager_get_mode", ret_cmp, 0);
+
+	TC_SUCCESS_RESULT();
+}
+
+
+static void utc_wifi_manager_connect_ap_n(void)
+{
+	wifi_manager_result_e ret = WIFI_MANAGER_FAIL;
+
+	/* Connect to AP */
+	wifi_manager_ap_config_s config = {\
+		"NETGEAR99", 9,\
+		"artikwifi", 9, \
+		WIFI_MANAGER_AUTH_WPA2_PSK, \
+		WIFI_MANAGER_CRYPTO_AES
+	};
+
+	/* current wifi mode is softap, then this try will fail */
+	ret = wifi_manager_connect_ap(&config);
+
+	TC_ASSERT_EQ("wifi_manager_connect_ap", ret, WIFI_MANAGER_FAIL);
+	TC_SUCCESS_RESULT();
+}
+
+static void utc_wifi_manager_connect_ap_p(void)
+{
+	wifi_manager_result_e ret = WIFI_MANAGER_FAIL;
+
+	/* change to station mode */
+	ret = wifi_manager_set_mode(STA_MODE, NULL);
+	TC_ASSERT_EQ("wifi_manager_connect_ap", ret, WIFI_MANAGER_SUCCESS);
+
+	/* Connect to AP */
+	wifi_manager_ap_config_s config = {\
+		"NETGEAR99", 9,\
+		"artikwifi", 9, \
+		WIFI_MANAGER_AUTH_WPA2_PSK, \
+		WIFI_MANAGER_CRYPTO_AES
+	};
+
+	/* current wifi mode is softap, then this try will fail */
+	ret = wifi_manager_connect_ap(&config);
+
+	TC_ASSERT_EQ("wifi_manager_connect_ap", ret, WIFI_MANAGER_SUCCESS);
+	TC_SUCCESS_RESULT();
+}
+
+static void utc_wifi_manager_deinit_p(void)
+{
+	wifi_manager_result_e ret = WIFI_MANAGER_FAIL;
+
+	ret = wifi_manager_deinit();
+
+	TC_ASSERT_EQ("wifi_manager_deinit", ret, WIFI_MANAGER_SUCCESS);
+	TC_SUCCESS_RESULT();
+}
+
+static void utc_wifi_manager_deinit_n(void)
+{
+	wifi_manager_result_e ret = WIFI_MANAGER_FAIL;
+
+	ret = wifi_manager_deinit();
+
+	TC_ASSERT_EQ("wifi_manager_deinit", ret, WIFI_MANAGER_DEINITIALIZED);
+	TC_SUCCESS_RESULT();
+}
+
+int wifi_manager_utc(int argc, FAR char *argv[])
+{
+	total_pass = 0;
```

Comment:
1. If you didn't use semaphore, next tc will be launched before finishing this. Let's use it like belows.
2. We should print the result of tc at end of function.
```
int kernel_tc_main(int argc, char *argv[])
{
	sem_wait(&tc_sem);
	working_tc++;

	total_pass = 0;
	total_fail = 0;

	printf("=== TINYARA Kernel TC START! ===\n");
...
...
	printf("\n=== TINYARA Kernel TC COMPLETE ===\n");
	printf("\t\tTotal pass : %d\n\t\tTotal fail : %d\n", total_pass, total_fail);

	working_tc--;
	sem_post(&tc_sem);

	return 0;
}
```

Reply:
Thanks for letting me know this. Done!